hack/check-latest-images.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# Copyright The Shipwright Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Check the latest image and replace
+
+# Usage
+# ./check-latest-image.sh <FILE> <IMAGE> <IMAGE_URL>
+
+set -euo pipefail
+
+# Check the number of arguments passed and exit if less than three
+if [ $# -lt 3 ]; then
+	echo "USAGE: $0 <FILE> <IMAGE> <IMAGE_URL>"
+	exit 1
+fi
+
+# Set environment variables
+IMAGE=$(printf '%s\n' "$(echo ${1})" | sed -e 's/[]\/$*.^[]/\\&/g')
+CURRENT_TAG=$(printf '%s\n' "$(grep -r -h -m 1 -o ${1}'[^ ]*' ${3} | head -n 1 | cut -d':' -f2)" | sed -e 's/[]\/$*.^[]/\\&/g')
+LATEST_TAG=$(printf '%s\n' "$(curl -s ${2} | jq -r '.name')" | sed -e 's/[]\/$*.^[]/\\&/g')
```

Comment:
You have a loop in the GitHub action with a curl against GitHub here. I've seen rate limitting happening and maybe that would happen here as well. So, I think you will have to move the for loop into this script.

Reply:
The `for` loop for retry?